### PR TITLE
[MNG-7097] Plugin Dependency Resolution Improvement

### PR DESCRIPTION
**Pull request by cstamas**
Implements a helper component for plugin dependency resolution that provider
filter and selector that leaves out core artifacts from collection
and resolution.

### CODE REVIEW

File: maven-core/src/main/java/org/apache/maven/extension/internal/CoreExportsProvider.java [hunk 1]
```diff
@@ -27,7 +27,8 @@
 import org.eclipse.sisu.Nullable;
 
 /**
- * CoreExportsProvider
+ * CoreExportsProvider, that CANNOT be <tty>javax.inject.Provider</tty> as ctor would have circular dep.
```
Comment (pzygielo) [inline maven-core/src/main/java/org/apache/maven/extension/internal/CoreExportsProvider.java:30]: `javadoc: error: unknown tag: tty`

Comment (cstamas) [reply maven-core/src/main/java/org/apache/maven/extension/internal/CoreExportsProvider.java]: Fixed, tx

**Comment by cstamas** [comment]
Something is strange: for example this one fails (on GH but also locally as well):
```
mng7045 DropUselessAndOutdatedCdiApi.it().................... FAILURE (0.3 s)
```
**but**, when I run the POM in IT from CLI (is simple, uses GMavenPlus plugin and a script), it passes ok...

**Comment by jira-importer** [comment]
Resolve #8359

**Pull request closed without merging**